### PR TITLE
refactor(pie-monorepo): DSW-1645 Remove Jest in favour of Vitest

### PR DESCRIPTION
**Pull request by siggerzz**
## Author Checklist (complete before requesting a review)
- [ ] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests
- [ ] If it is a `PIE Docs` change, I have reviewed the Docs site preview
- [ ] If it is a component change, I have reviewed the Storybook preview
- [ ] If there are visual test updates, I have reviewed them properly before approving

### Reviewer 1
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### Reviewer 2
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### CODE REVIEW

File: apps/examples/wc-angular12/package.json [hunk 1]
```diff
@@ -27,13 +27,6 @@
         "@angular-devkit/build-angular": "15.2.6",
         "@angular/cli": "15.2.6",
         "@angular/compiler-cli": "15.2.0",
-        "@types/jasmine": "4.3.0",
```
Comment (siggerzz) [inline apps/examples/wc-angular12/package.json:30]: This change was done to remove unneeded dependencies, given we currently don't unit test Angular.

The main reason for this, was due to the fact I want to make the Vitest types available globally, and given most test libraries follow the `describe` / `it` / `expect` convention, the addition of `@types/jasmine` was causing type conflicts.

File: configs/pie-components-config/tsconfig.json [hunk 2]
```diff
@@ -21,6 +21,8 @@
     "forceConsistentCasingInFileNames": true,
     "allowImportingTsExtensions": true,
     "noEmit": true,
+    "skipLibCheck": true,
```
Comment (siggerzz) [inline configs/pie-components-config/tsconfig.json:24]: As part of this refactor, I had to upgrade the version of Vite that we're using in the repo.

Unfortunately, there's currently an issue with some of the Vite declarations, that mean when building certain components, the following is output in the console (although the build does still succeed):

```bash
@justeattakeaway/pie-tag:build: [vite:dts] Start generate declaration files...
@justeattakeaway/pie-tag:build: ../../../node_modules/@vitest/utils/dist/error.d.ts:4:58 - error TS2304: Cannot find name 'WeakKey'.
@justeattakeaway/pie-tag:build: 
@justeattakeaway/pie-tag:build: 4 declare function serializeError(val: any, seen?: WeakMap<WeakKey, any>): any;
@justeattakeaway/pie-tag:build:                                                            ~~~~~~~
@justeattakeaway/pie-tag:build: ../../../node_modules/@vitest/utils/dist/error.d.ts:6:96 - error TS2304: Cannot find name 'WeakKey'.
@justeattakeaway/pie-tag:build: 
@justeattakeaway/pie-tag:build: 6 declare function replaceAsymmetricMatcher(actual: any, expected: any, actualReplaced?: WeakSet<WeakKey>, expectedReplaced?: WeakSet<WeakKey>): {
@justeattakeaway/pie-tag:build:                                                                                                  ~~~~~~~
@justeattakeaway/pie-tag:build: ../../../node_modules/@vitest/utils/dist/error.d.ts:6:133 - error TS2304: Cannot find name 'WeakKey'.
@justeattakeaway/pie-tag:build: 
@justeattakeaway/pie-tag:build: 6 declare function replaceAsymmetricMatcher(actual: any, expected: any, actualReplaced?: WeakSet<WeakKey>, expectedReplaced?: WeakSet<WeakKey>): {
@justeattakeaway/pie-tag:build:                                                                                                                                       ~~~~~~~
@justeattakeaway/pie-tag:build: ../../../node_modules/vitest/node_modules/vite/dist/node/index.d.ts:6:41 - error TS2307: Cannot find module 'rollup/parseAst' or its corresponding type declarations.
@justeattakeaway/pie-tag:build: 
@justeattakeaway/pie-tag:build: 6 export { parseAst, parseAstAsync } from 'rollup/parseAst';
```

Having `skipLibCheck` set, removes these errors, but I appreciate it's not ideal. As noted in the [Typescript docs](https://www.typescriptlang.org/tsconfig#skipLibCheck), this is less type-safe.

When we run `yarn why vite`, we see a lot of versions being used throughout ther repo:

```
├─ @nuxt/vite-builder@npm:3.4.3
│  └─ vite@npm:4.3.9 (via npm:~4.3.3)
│
├─ @nuxt/vite-builder@npm:3.4.3 [5f25d]
│  └─ vite@npm:4.3.9 [b37cd] (via npm:~4.3.3 [b37cd])
│
├─ @playwright/experimental-ct-react@npm:1.32.3
│  └─ vite@npm:4.5.2 [9b7b3] (via npm:^4.1.1 [9b7b3])
│
├─ pie-monorepo@workspace:.
│  └─ vite@npm:5.0.12 [c9f5e] (via npm:5.0.12 [c9f5e])
│
├─ vite-node@npm:0.30.1
│  └─ vite@npm:4.5.2 [9b7b3] (via npm:^4.1.1 [9b7b3])
│
├─ vite-node@npm:1.2.1
│  └─ vite@npm:5.0.12 [047b3] (via npm:^5.0.0 [047b3])
│
├─ vitest@npm:1.2.1
│  └─ vite@npm:5.0.12 (via npm:^5.0.0)
│
├─ vitest@npm:1.2.1 [c9f5e]
│  └─ vite@npm:5.0.12 [49894] (via npm:^5.0.0 [49894])
│
├─ wc-vanilla@workspace:apps/examples/wc-vanilla
│  └─ vite@npm:4.5.2 [9b7b3] (via npm:^4.1.1 [9b7b3])
│
└─ wc-vue3@workspace:apps/examples/wc-vue3
   └─ vite@npm:4.5.2 [69a92] (via npm:4.5.2 [69a92])
```

I suspect the underlying issue is caused by different packages using different major versions.

More info can be found here, but I think we should leave this enabled for now, and look to turn it off once Vite 5 has matured a bit. This 'skipLibCheck' is also done in a number of other places in the repo, hence why I opted to get around this issue this way for the time being.

File: apps/pie-docs/src/__tests__/_11ty/shortcodes/categorisedIconList.test.js [hunk 3]
```diff
@@ -2,7 +2,7 @@ const categorisedIconList = require('../../../_11ty/shortcodes/categorisedIconLi
 
 describe('categorisedIconList.js', () => {
     // This is a temp fix until we look into resolving the icon / changeset issue properly.
-    xit('should return the expected HTML', () => {
+    it('should return the expected HTML', () => {
```
Comment (xander-marjoram) [inline apps/pie-docs/src/__tests__/_11ty/shortcodes/categorisedIconList.test.js:5]: If you're reenabling these tests I guess the above comments can be removed?

File: apps/pie-docs/src/__tests__/_utilities/navigation.test.js [hunk 4]
```diff
@@ -5,6 +5,6 @@ describe('PIE Docs - Navigation', () => {
     it('should compare the actual navigation routes to the JSON snapshot.', () => {
         const routes = getNavigationRoutes();
 
-        expect(routes, 'If you believe this route is expected, please run yarn test:generate-routes to regenerate the expected routes snapshot.').toEqual(expectedRoutes);
```
Comment (xander-marjoram) [inline apps/pie-docs/src/__tests__/_utilities/navigation.test.js:8]: What was wrong with this? The failure message looks helpful to me

Comment (siggerzz) [reply apps/pie-docs/src/__tests__/_utilities/navigation.test.js:8]: After adding the Vitest eslint rules, it was saying to remove this. It looks like it's not even a valid function argument!

File: configs/pie-components-config/tsconfig.json (hunk 2 above)
Comment (siggerzz) [reply configs/pie-components-config/tsconfig.json]: Mainly the fact our version a year out of date and we had some deprecation warnings. I've reverted this now.

File: apps/pie-docs/src/__tests__/_utilities/__snapshots__/colors.test.js.snap [hunk 5]
```diff
@@ -1,39 +1,39 @@
-// Jest Snapshot v1, https://goo.gl/fbAQLP
+// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html
 
-exports[`colors.js convertHexcodeToRBG converts hexcode "#000" to the expected RGB object 1`] = `
-Object {
+exports[`colors.js > convertHexcodeToRBG > converts hexcode %p to the expected RGB object #000 1`] = `
```
Comment (xander-marjoram) [inline apps/pie-docs/src/__tests__/_utilities/__snapshots__/colors.test.js.snap:3]: This still says `RBG`

Comment (siggerzz) [reply apps/pie-docs/src/__tests__/_utilities/__snapshots__/colors.test.js.snap:3]: That's the same as before. The only bit that was incorrect was the function name, not the test name :)

**Review by jamieomaguire — CHANGES_REQUESTED** [review]
as discussed it would be great to split this into multiple PRs :)

File: apps/pie-docs/src/__tests__/_utilities/__snapshots__/colors.test.js.snap (hunk 5 above)
Comment (xander-marjoram) [reply apps/pie-docs/src/__tests__/_utilities/__snapshots__/colors.test.js.snap:3]: But the test name must also be incorrect if it still says RBG 😄

Comment (xander-marjoram) [reply apps/pie-docs/src/__tests__/_utilities/__snapshots__/colors.test.js.snap:3]: Ah I see, if you've undone the other corrections then that's fine

File: apps/pie-docs/src/__tests__/_11ty/shortcodes/__snapshots__/card.test.js.snap [hunk 6]
```diff
@@ -22,43 +22,43 @@ exports[`card.js should return the expected HTML 2`] = `
...
+                <svg xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"presentation\\" focusable=\\"false\\" fill=\\"#000\\" viewBox=\\"0 0 16 16\\" height=\\"21\\" width=\\"21\\" aria-hidden=\\"true\\" class=\\"c-pieIcon c-pieIcon--linkExternal\\"><path d=\\"M13.25 3.564 8.21 8.63l-.928-.928 5.093-5.066H9.662V1.324h3.833a1.093 1.093 0 0 1 1.067 1.093V6.25H13.25V3.564Z\\"></path><path d=\\"M11.302 12.831c.343-.34.537-.804.54-1.287V7.598h1.312v3.946a3.141 3.141 0 0 1-3.142 3.132h-5.46a3.141 3.141 0 0 1-3.14-3.132V6.075a3.132 3.132 0 0 1 3.14-3.132h3.79v1.312h-3.79a1.829 1.829 0 0 0-1.802 1.82v5.469a1.829 1.829 0 0 0 1.802 1.82h5.46c.484 0 .948-.192 1.29-.533Z\\"></path></svg>
```
Comment (jamieomaguire) [inline apps/pie-docs/src/__tests__/_11ty/shortcodes/__snapshots__/card.test.js.snap:42]: This appears to be adding some new attributes to the SVG is there a newer version of the icon being used or something?

Also just double checking that it seems like some of the path `fill` attributes have changed, is this expected?

**Review by jamieomaguire — APPROVED** [review]

**Review by JoshuaNg2332 — APPROVED** [review]

File: apps/pie-docs/src/__tests__/_11ty/shortcodes/__snapshots__/card.test.js.snap (hunk 6 above)
Comment (xander-marjoram) [reply apps/pie-docs/src/__tests__/_11ty/shortcodes/__snapshots__/card.test.js.snap:42]: What was the resolution on this?

**Pull request merged**